### ISSUE TITLE
[109] 🦩 Flamingo: a Visual Language Model for Few-Shot Learning

### ISSUE THREAD

**Issue by long8v** [long8v/PTIR#118]
<img width="762" alt="image" src="https://user-images.githubusercontent.com/46675408/230816882-9ca1b993-82ac-4408-b481-5893ff3ab715.png">

[paper](https://arxiv.org/pdf/2204.14198.pdf)

## TL;DR
- **I read this because.. :** #116 읽고 나서 읽고 싶어짐. 예전에 성현님이 소개해주셨는데 디테일 모름. GPT 열풍으로 요즘 다시 많이 언급됨.
- **task :** Vision Language Model in general use!
- **problem :** CLIP 류는 image-text pair의 score만 나오므로 이미지 분류와 같은 closed set에 대한 태스크에만 적용이 가능하다. cpationing이나 VQA 같은 open-ended task를 풀 수 있는 generate language 능력이 부족하다. 
- **idea :**
- **architecture :**
- **objective :**
- **baseline :**
- **data :** 
- **evaluation :**
- **result :**
- **contribution :**
- **limitation / things I cannot understand :**

## Details
- ECCV workshop 때 Jean-Baptiste가 flamingo를 하게 된 이유 / 하면서 느낀 점을 소개해준 적이 있음
<img width="200" alt="image" src="https://user-images.githubusercontent.com/46675408/230821196-a747f1c7-7416-482c-b7c5-caef8a119b6d.png">

introduction에 써있는거랑 비슷한 내용. CLIP류 연구를 했었는데 풀 수 있는 task가 한정적이었다. -> flamingo로 넘어감
결국 어떤 인터페이스가 다양한 태스크를 풀 수 있을 것인가? application에 적합할 것인가?를 문제의식으로 삼은 것 같당
문제 의식을 아키텍쳐가 아니라 풀 수 있는 task 들로 잡은 듯~ 흠 점점 아키텍쳐가 중요한게 아니라 데이터/학습/태스크 등이 중요한 것 같네.. 나는 이제 무얼 쌓아야 하나 

### Preliminaries 
- Normalizer Free ResNet
https://arxiv.org/pdf/2102.06171.pdf
ResNet의 batch norm이 모델이 bs에 민감해지거나, 한 배치 내 이미지의 interaction에 영향을 받게 하는 효과가 있어서 이를 해결하기 위한 모델\

- Perceiver
https://arxiv.org/pdf/2103.03206.pdf
<img width="680" alt="image" src="https://user-images.githubusercontent.com/46675408/230818470-037388bc-cec2-465b-8a53-f5350bdfc903.png">

21년도 deep mind에서 image / video 등 다양한 modality를 효율적으로 표현할 수 있게.
비대칭적인 attention 모듈을 사용해서 a small set of latent units으로 점차 CA할 수 있도록(detr이랑 비슷한데 디테일이 좀 다를듯)
image classification / audio / point cloud 등에서 comparable 성능
(c.f. Set Transformer가 most related work라고 하면서 계속 언급)

c.f. x-attn에서 x가 뭐지 하고 검색하다 발견
전체 finetuning 안하고 CA쪽만 해도 성능이 좋다는 논문. domain은 MT
Cross-Attention is All You Need:  Adapting Pretrained Transformers for Machine Translation
https://arxiv.org/pdf/2104.08771.pdf

**Comment by long8v** [comment]
논문 스터디에서 발표함.
[google slide](https://docs.google.com/presentation/d/10Rz1Bg6xLr4n7q_CHV6Ku5M-O4voh-nMlfivJHDqNzE/edit?usp=sharing)에 더 많은 디테일.
